allennlp/data/data_loaders/multi_process_data_loader.py

```diff
@@ -123,14 +168,23 @@ class MultiProcessDataLoader(DataLoader):
 
         See [issue #4848](https://github.com/allenai/allennlp/issues/4848) for more info.
 
-    !!! Warning
-        Another issue besides dead-locks that you could run into when using `num_workers > 0`
-        is running out of shared memory, since tensors are passed between processes
-        using shared memory, and some systems impose strict limits on the allowed size of shared
-        memory.
+        Dead-locks could also be caused by running out of shared memory (see below).
+
+    - **Shared memory restrictions**
 
-        Luckily there is also a simple work-around for this. Either decrease `max_instances_in_memory`
-        or increase your system's `ulimit`.
+        Tensors are passed between processes using shared memory, and some systems impose strict
+        limits on the allowed size of shared memory.
+
+        Luckily this is simple to debug and simple to fix.
+
+        First, to verify that this is your issue just watch your shared memory as your data loader runs.
+        For example, run `watch -n 0.3 'df -h | grep shm'`.
+
+        If you're seeing your shared memory blow up until it maxes-out, then you either need to decrease
+        `max_instances_in_memory` or increase your system's `ulimit`.
+
+        If you're using Docker, you can increase the shared memory available on a container by running
+        it with the option `--ipc=host` or by setting `--shm-size`.
```

Comment:
This is some heavy stuff to keep in your head :-/

Reply:
That's why I wrote it down 😜

Reply:
Well, that's better than not having it written down.

But our lofty ideal should be that data loading is something a researcher never even thinks about. It just does the right thing automatically. 

I'm not sure we can get there with the tools we have right now (`TOKENIZERS_PARALLELISM` what?), and this is definitely a step up. I just don't like that data loading is so complicated in general.